docs/development/zod.md

```diff
@@ -0,0 +1,232 @@
+# Zod schema guideline
+
+Renovate has adopted Zod for schema validation and it is desirable that any new manager or datasource work uses this approach.
+This document describes some guides as to how and why to use Zod features.
+
+The key concept of schema validation is to find the right balance between strictness of contracts between separately developed systems and the permissiveness of the Renovate itself.
+We want Renovate to be only as strict as it needs to be (e.g. about optional fields which a public registry might always have but self-hosted implementations may leave off) but not to make _assumptions_ about the presence of fields which could lead to run-time errors when they are missing.
+
+## When/where to use Zod
+
+We should use Zod for validating:
+
+- Data received from external APIs and data sources, particularly the `lib/modules/datasource/*` section of Renovate
+- Data parsed from files in the repository, particularly the `lib/modules/manager/*` section of Renovate
+
+The `cdnjs` datasource is a good example of using Zod schema validations on API responses from external sources.
+
+The `composer` manager is a good example of using Zod schema validation in a manager to validate parsed files in a repository.
+
+## Technical guide
+
+### Use schema.ts files for Zod schemas
+
+Try to locate/isolate Zod schemas in their own schema.ts files to keep them organized and reusable. [TODO: do our examples follow that?]
```

Comment:
We need to pick other examples or _improve_ our examples if not

Reply:
I've added a paragraph stating something like "while it's the general rule, sometimes it can be broken when it makes sense"

Reply:
Can't we get _one_ example?